packages/api/src/util/ensembleService.ts

```diff
@@ -0,0 +1,256 @@
+/*
+ *  Copyright 2025 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This file extends the Ensemble service (ensemble.js) using TypeScript. When the TS migration of ensemble.js is done, the contents of this file can be moved there.
+ */
+import { AxiosError } from 'axios';
+import FarmAddonModel from '../models/farmAddonModel.js';
+import AddonPartnerModel from '../models/addonPartnerModel.js';
+import LocationModel from '../models/locationModel.js';
+import ManagementPlanModel from '../models/managementPlanModel.js';
+import { customError } from './customErrors.js';
+import { ENSEMBLE_BRAND, ensembleAPI, ensembleAPICall } from './ensemble.js';
+
+interface Point {
+  lat: number;
+  lng: number;
+}
+
+enum PlantingMethod {
+  BED_METHOD = 'bed_method',
+  CONTAINER_METHOD = 'container_method',
+  BROADCAST_METHOD = 'broadcast_method',
+  ROW_METHOD = 'row_method',
+}
+
+type MethodDetails = {
+  planting_management_plan_id?: string;
+};
+
+interface PlantingManagementPlan {
+  planting_method: PlantingMethod;
+  bed_method: MethodDetails | null;
+  container_method: MethodDetails | null;
+  broadcast_method: MethodDetails | null;
+  row_method: MethodDetails | null;
+}
+
+interface ManagementPlan {
+  management_plan_id: string;
+  crop_management_plan: {
+    seed_date: string;
+    planting_management_plans: PlantingManagementPlan[];
+  };
+  crop_variety: {
+    crop: {
+      crop_common_name: string;
+      crop_genus: string;
+      crop_specie: string;
+    };
+  };
+}
+
+export interface LocationAndCropGraph {
+  farm_id: string;
+  name: string;
+  location_id: string;
+  figure: {
+    area: {
+      grid_points: Point[];
+    };
+  };
+  management_plans: ManagementPlan[];
+}
+
+interface EnsembleCropData {
+  crop_common_name: string;
+  crop_genus: string;
+  crop_specie: string;
+  seed_date: string;
+  planting_details?: {
+    planting_method: PlantingMethod;
+    [key: string]: unknown;
+  };
+}
+
+export interface EnsembleLocationAndCropData {
+  farm_id: string;
+  name: string;
+  location_id: string;
+  grid_points: Point[];
+  crop_data: EnsembleCropData[];
+}
+
+interface OrganisationFarmData {
+  [org_uuid: string]: EnsembleLocationAndCropData[];
+}
```

Comment:
Maybe we could create a separate types file to keep this one shorter and have all the types bundled together?